app/Models/Post.php

```diff
@@ -157,4 +155,39 @@ public function toBlinkPayload()
             'deleted_at' => $this->deleted_at ? $this->deleted_at->toIso8601String() : null,
         ];
     }
+
+    /**
+     * Apply the given tag to this post.
+     */
+    public function tag($tagName)
+    {
+        $tag = Tag::where('tag_name', $tagName)->first();
+
+        // Create the tag if it doesn't already exist (so we don't have to create tags separately)
+        if (! $tag) {
+            $tag = Tag::create(['tag_name' => $tagName, 'tag_slug' => str_slug($tagName, '-')]);
+        }
+
+        $this->tags()->attach($tag);
+
+        // Update timestamps on the Post when adding a tag
+        $this->touch();
+
+        return $this;
+    }
+
+    /**
+     * Remove the given tag from this post.
+     */
+    public function untag($tagName)
+    {
+        $tag = Tag::where('tag_name', $tagName)->first();
```

Comment:
@DFurnes I think we eventually want to use the slug for everything, but I didn't change to that here to not change too much at once. The problem with that is when a new tag is created, could potentially be weird getting the name from the slug?